src/ProjectOrigin.WalletSystem.IntegrationTests/GrpcTests.cs

```diff
@@ -0,0 +1,72 @@
+using Dapper;
+using FluentAssertions;
+using Grpc.Core;
+using ProjectOrigin.WalletSystem.IntegrationTests.TestClassFixtures;
+using ProjectOrigin.WalletSystem.Server;
+using ProjectOrigin.WalletSystem.Server.Database;
+using ProjectOrigin.WalletSystem.Server.Models;
+using ProjectOrigin.WalletSystem.V1;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace ProjectOrigin.WalletSystem.IntegrationTests;
+
+public class GrpcTests : GrpcTestsBase
+{
+    private readonly JwtGenerator _tokenGenerator;
+
+    public GrpcTests(GrpcTestFixture<Startup> grpcFixture, PostgresDatabaseFixture dbFixture, ITestOutputHelper outputHelper) : base(grpcFixture, dbFixture, outputHelper)
+    {
+        _tokenGenerator = new JwtGenerator();
+    }
+
+    [Fact]
+    public async Task CreateWalletSection_ValidResponse_InDatabase()
+    {
+        // Arrange
+        var subject = Guid.NewGuid().ToString();
+        var token = _tokenGenerator.GenerateToken(subject, "John Doe");
+
+        var headers = new Metadata { { "Authorization", $"Bearer {token}" } };
+
+        var client = new WalletService.WalletServiceClient(_grpcFixture.Channel);
+        var request = new CreateWalletSectionRequest();
+
+        // Act
+        var walletSection = await client.CreateWalletSectionAsync(request, headers);
+
+        // Assert
+        walletSection.Should().NotBeNull();
+        walletSection.Version.Should().Be(1);
+        walletSection.Endpoint.Should().Be(endpoint);
+        walletSection.SectionPublicKey.Should().NotBeNullOrEmpty();
+
+        using (var connection = new DbConnectionFactory(_dbFixture.ConnectionString).CreateConnection())
+        {
+            var foundSection = connection.QuerySingle<WalletSection>("SELECT * FROM WalletSections");
+
+            walletSection.SectionPublicKey.Should().Equal(foundSection.PublicKey.Export().ToArray());
+
+            var foundWallet = connection.QuerySingle<Wallet>("SELECT * FROM Wallets where owner = @owner", new { owner = subject });
+            // Wallet should be implicitly created
+            foundWallet.Should().NotBeNull();
+        }
```

Comment:
I disagree, if the test fails, then I would be unsure which of the two API calls that failed, and secondly, there is no API to get this data.

An alternative solution would likely be to implement an stub repository and use that instead of the real repositories, and remove the database entirely when doing gRPC integration tests.

Reply:
I would prefer here to have a unit test that checks for the side effect of creating a wallet if it does not exist. The unit test should use a stubbed repository. 

It is possible to replace with stub repositories in the gRPC integration tests, but I think that would make them less trustworthy and will not catch mis-configuration or wrong usage of the database.

Reply:
The test is more a replacement for unit-tests and not a full flow test.